python/lsst/ts/watcher/rules/base_ess_rule.py

```diff
@@ -0,0 +1,278 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["BaseEssRule"]
+
+import asyncio
+
+from lsst.ts.idl.enums.Watcher import AlarmSeverity
+from lsst.ts import utils
+from lsst.ts import watcher
+
+
+class BaseEssRule(watcher.PollingRule):
+    """Check one kind of ESS data, e.g. temperatur or humidity.
+
+    Parameters
+    ----------
+    config : struct
+        Rule configuration. The following fields are read:
+
+        * warning_level : `float` | `None`
+            Warning level. None to not use this level.
+        * serious_level : `float` | `None`
+            Serious level. None to not use this level.
+        * critical_level : `float` | `None`
+            Critical level. None to not use this level.
+        * hysteresis : `float`
+            Hysteresis.
+
+        * sensor_info: a list of dicts whose format depends on is_indexed.
+          If is_indexed false, the fields are:
+
+          * sal_index (int): SAL index of topic
+          * sensor_names (list[str]): a list of sensor names.
+
+          If is_indexed true, the fields are:
+
+          * sal_index (int): SAL index of topic
+          * sensor_info (dict) : a list of topic-specific sensor info.
+            Each item contains these fields:
+
+            * sensor_name (str): name of sensor
+            * indices (list[int], optional): indices of field to read,
+              If omitted then read all indices.
+
+    rule_name : str
+        The name of the rule.
+    topic_attr_name : str
+        The attr name of the ESS telemetry topic, e.g. "tel_temperature"
+        or "tel_relativeHumidity".
+    field_name : str
+        The name of the ESS topic field, e.g. "temperature" or "humidity".
+    is_indexed : bool
+        Is the field indexed? This controls the format of sensors.
+    min_config : struct | None
+        Configuration for minimum values.
+        The following fields are read, if present:
+
+        * warning_level
+        * serious_level
+        * critical_level
+
+    max_config : struct | None
+        Configuration for maximum values
+        The following fields are read, if present:
+
+        * warning_level
+        * serious_level
+        * critical_level
+
+    sensors : list[dict]
+        A list of sensor info, whose format depends on is_indexed.
+        If is_indexed true then the fields are:
+
+        * sal_index (int): SAL index of topic
+        * sensor_info (dict) : a list of topic-specific sensor info.
+          Each item contains these fields:
+
+          * sensor_name (str): name of sensor
+          * indices (list[int], optional): indices of field to read,
+            If omitted then read all indices.
```

Comment:
Please either terminate the previous sentence in `.` instead of `,` or start this sentence with an `i` instead of an `I`.